azure/scope/machine.go

```diff
@@ -221,6 +222,73 @@ func (m *MachineScope) InboundNatSpecs(portsInUse map[int32]struct{}) []azure.Re
 
 // NICSpecs returns the network interface specs.
 func (m *MachineScope) NICSpecs() []azure.ResourceSpecGetter {
+	nicSpecs := []azure.ResourceSpecGetter{}
+	defaultSpec := m.DefaultNICSpec()
+	if len(m.AzureMachine.Spec.NetworkInterfaces) < 1 {
+		return append(nicSpecs, defaultSpec)
+	}
+
+	for i, n := range m.AzureMachine.Spec.NetworkInterfaces {
+		var spec *networkinterfaces.NICSpec
+		if n.ID != "" {
+			continue
+		}
+		if i == 0 {
+			spec = m.DefaultNICSpec()
```

Comment:
it's unclear to me why we're using the defaults for index 0 but not for the others. What's significant about the order the network interfaces are specified in? What's different between the default spec and the fields we're setting below in the `else` statement?